pkg/internal/checkup/checkup_test.go

```diff
@@ -149,6 +159,9 @@ func TestTeardownShouldFailWhen(t *testing.T) {
 			assert.NoError(t, testCheckup.Setup(context.Background()))
 			assert.NoError(t, testCheckup.Run(context.Background()))
 
+			testClient.vmiDeletionFailure = testCase.vmiDeletionFailure
+			testClient.vmiReadFailure = testCase.vmiReadFailure
+			testClient.podDeletionFailure = testCase.podDeletionFailure
```

Comment:
Could you please move the error injection after the `testClient` initialization?
I think it makes the tests more readable.

Reply:
the error should be inserted only before the Teardown, because otherwise we could hit the error sooner.
For example, the pod is read during both Setup and Teardown..

Logically speaking - this is where these errors should be inserted, cause that's the point in time I want to inject them in